plugins/uefi-dbx/fu-uefi-dbx-plugin.c

```diff
@@ -48,19 +86,74 @@ fu_uefi_dbx_plugin_init(FuUefiDbxPlugin *self)
 {
 }
 
+static void
+fu_uefi_dbx_plugin_finalize(GObject *object)
+{
+#ifdef WITH_UEFI_DBX_SNAPD_NOTIFIER
+	FuUefiDbxPlugin *self = FU_UEFI_DBX_PLUGIN(object);
+	if (self->snapd_notifier != NULL) {
+		g_object_unref(self->snapd_notifier);
+		self->snapd_notifier = NULL;
+	}
+#endif
+	G_OBJECT_CLASS(fu_uefi_dbx_plugin_parent_class)->finalize(object);
+}
+
 static void
 fu_uefi_dbx_plugin_constructed(GObject *obj)
 {
 	FuPlugin *plugin = FU_PLUGIN(obj);
 	fu_plugin_add_rule(plugin, FU_PLUGIN_RULE_METADATA_SOURCE, "uefi_capsule");
 	fu_plugin_add_firmware_gtype(plugin, NULL, FU_TYPE_EFI_SIGNATURE_LIST);
 	fu_plugin_add_device_gtype(plugin, FU_TYPE_UEFI_DBX_DEVICE);
+
+#ifdef WITH_UEFI_DBX_SNAPD_NOTIFIER
+	if (fu_snap_is_in_snap()) {
+		/* only enable snapd integration if running inside a snap */
+		if (fu_uefi_dbx_plugin_snapd_notify_init(FU_UEFI_DBX_PLUGIN(obj))) {
+			g_info("snapd integration enabled ");
+		} else {
+			g_info("snapd integration disabled");
+		}
+	} else {
+		/* TODO figure out non-snap scenarios */
+		g_info("snapd integration outside of snap is not supported");
+	}
+#endif /* WITH_UEFI_DBX_SNAPD_NOTIFIER */
 }
 
 static void
 fu_uefi_dbx_plugin_class_init(FuUefiDbxPluginClass *klass)
 {
+	GObjectClass *object_class = G_OBJECT_CLASS(klass);
 	FuPluginClass *plugin_class = FU_PLUGIN_CLASS(klass);
+
 	plugin_class->constructed = fu_uefi_dbx_plugin_constructed;
 	plugin_class->coldplug = fu_uefi_dbx_plugin_coldplug;
+
+	object_class->finalize = fu_uefi_dbx_plugin_finalize;
+}
+
+#ifdef WITH_UEFI_DBX_SNAPD_NOTIFIER
+static gboolean
+fu_uefi_dbx_plugin_snapd_notify_init(FuUefiDbxPlugin *self)
```

Comment:
Could we pass in a GError? I think handling the error in the caller rather than locally would feel more natural.